### PR TITLE
Add probe

### PR DESCRIPTION
**Pull request by opencmit2**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

Modify the minio chart to add liveness and readiness probes for the minio container. 

## Motivation and Context

Although minio takes a long time to start up, this shouldn't prevent adding this functionality. Health checks are disabled by default.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: helm/minio/templates/statefulset.yaml [hunk 1]
```diff
@@ -131,8 +131,48 @@ spec:
...
+              protocol: TCP
             - name: {{ $scheme }}-console
               containerPort: {{ .Values.minioConsolePort }}
+              protocol: TCP
+          {{- if .Values.customLivenessProbe }}
+          livenessProbe: {{- include "minio.tplvalues.render" (dict "value" .Values.customLivenessProbe "context" $) | nindent 12 }}
+          {{- else if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /minio/health/live
```
Comment (jiuker) [inline helm/minio/templates/statefulset.yaml:143]: `/minio/health/live` depend on status of other applications. We can't kill minio for some applications are offline. And if minio is `busy` , we can't kill minio also that will cause the minio-cluster unavailable.  And We never kill minio.

**Review by jiuker — CHANGES_REQUESTED** [review]

File: helm/minio/templates/statefulset.yaml (hunk 1 above)
Comment (opencmit2) [reply helm/minio/templates/statefulset.yaml:143]: So I understand this /minio/health/live interface, but I can't tell whether minio is abnormal?

Comment (jiuker) [reply helm/minio/templates/statefulset.yaml:143]: Yeah. Not available for now.

Comment (opencmit2) [reply helm/minio/templates/statefulset.yaml:143]: Are there other interfaces available?

Comment (jiuker) [reply helm/minio/templates/statefulset.yaml:143]: None. We never kill minio.

**Comment by adlnc** [comment]
Since this https://github.com/minio/minio/pull/18457 as been merged last week, does it make sense to consider re-opening this one ?

**Comment by sangheee** [comment]
Please reconsider this change

**Pull request closed without merging**